src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserBase.cs

```diff
@@ -0,0 +1,193 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Globalization;
+using System.Linq;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+
+namespace Microsoft.SqlTools.ServiceLayer.FileBrowser
+{
+    public struct FileInfo
+    {
+        // Empty for folder
+        public string fileName;
+        public string folderName;
+        public string path;
+        // Includes file name in the path. Empty for folder.
+        public string fullPath;
+    }
+
+    /// <summary>
+    /// Base class for file browser
+    /// </summary>
+    public abstract class FileBrowserBase
+    {
+        private Enumerator enumerator = null;
+        protected object sqlConnection = null;
+
+        protected Enumerator Enumerator
+        {
+            get
+            {
+                return this.enumerator = (this.enumerator ?? new Enumerator());
+            }
+        }
+
+        /// <summary>
+        /// Separator string for components of the file path. Defaults to \ for Windows and / for Linux
+        /// </summary>
+        internal string PathSeparator { get; set; }
+
+        /// <summary>
+        /// Returns the PathSeparator values of the Server.
+        /// </summary>
+        /// <returns>PathSeparator</returns>
+        internal static string GetPathSeparator(Enumerator sfcEnumerator, object sqlConnectionObject)
+        {
+            var req = new Request();
+            req.Urn = "Server";
+            req.Fields = new[] { "PathSeparator" };
+
+            DataSet ds = sfcEnumerator.Process(sqlConnectionObject, req);
```

Comment:
this will cause you trouble when two requests getting to this line at the same time

Reply:
This will only be a problem if it gets multiple requests for the same connection. But the services that use filebrowser service should pass unique connection. For example, backup creates a unique connection each time opening a new dialog and it passes the uri to the file browser. This way, the file browser operations will be executed on its dedicated connection.